sycl/include/CL/sycl/detail/scheduler/commands.hpp

```diff
@@ -116,10 +116,16 @@ class Command {
 
   std::shared_ptr<event_impl> getEvent() const { return MEvent; }
 
-  virtual ~Command() = default;
-
   virtual void printDot(std::ostream &Stream) const = 0;
 
+  virtual const Requirement *getRequirement() const {
+    throw sycl::runtime_error(
+        "Internal Error. The command has no stored requirement",
+        PI_INVALID_OPERATION);
```

Comment:
Sorry for missing this the first time.
Shouldn't this be an assertion?

Reply:
Why?

Reply:
Because it says: "Internal Error."
I assume it's not something user can catch and recover from. Right?
It looks like it's useful for run-time library developers to make sure that this method is overridden by child classes.
Assertions are the right mechanism for detecting internal API inconsistencies like this one.

Reply:
Done.